scripts: runners: improvements for spi_burn west runner

rikorsev opened the pull request:
Following changes has been made in this PR:
    - fix issue when a binary is not able to be found by relative path
    - implement "west debug" command
    - implement --dt-flash option

rikorsev committed:
scripts: runners: improvements for spi_burn west runner

Following changes has been made by this commit:
    - fix issue when a binary is not able to be found by relative path
    - implement "west debug" command
    - implement --dt-flash option

Signed-off-by: Alex Kolosov <rikorsev@gmail.com>

rikorsev commented:
Hi @mbolivar-nordic,

May I ask you to review the PR?

Regards,
Alex

rikorsev commented:
Hi @nashif, @kgugala, @pgielda,

Please, review the PR

Regards,
Alex

mbolivar-nordic commented on boards/common/spi_burn.board.cmake:
```diff
@@ -1,4 +1,5 @@
 board_set_flasher_ifnset(spi_burn)
-board_finalize_runner_args(spi_burn)
+board_set_debugger_ifnset(spi_burn)
+board_finalize_runner_args(spi_burn)
```
Please add missing newline at end of file.

mbolivar-nordic commented on boards/riscv/tlsr9518adk80d/doc/index.rst:
```diff
@@ -255,7 +255,23 @@ It is also possible to use the west flash command, but additional steps are requ
 =========
 
-Supporting UART debug and OpenOCD+GDB.
+Supporting UART debug and OpenOCD+GDB. west debug command also supported. You may run it in a simle way, like:
```
```suggestion
This port supports UART debug and OpenOCD+GDB. The `west debug` command also supported. You may run it in a simple way, like:
```

rikorsev replied:
Done

mbolivar-nordic reviewed (commented):
I'm sorry, I thought I had submitted my review and I lost track of this during the v3.1 release. Here are some initial comments.  Thanks for your patch.

rikorsev committed:
scripts: runners: extend popen_ignore_init method with **kwargs agrument

Extend popen_ignore_init with **kwargs to provide opportunity to pass
additional named arguments in subprocesses.Popen call

Signed-off-by: Alexandr Kolosov <rikorsev@gmail.com>

rikorsev commented on boards/riscv/tlsr9518adk80d/doc/index.rst:
```diff
@@ -255,7 +255,20 @@ It is also possible to use the west flash command, but additional steps are requ
+   west debug --gdb-port=<port_number> --gdb-ex=<additional_ex_arguments>
+
+Example:
+   west debug --gdb-port=1111 --gdb-ex="-ex monitor reset halt -ex b main -ex continue"
```
Done

rikorsev commented on scripts/west_commands/runners/spi_burn.py:
```diff
@@ -59,14 +71,49 @@ def do_run(self, command, **kwargs):
+        # which is necessary for correct working of GDB.
+        # The part is copied from popen_ignore_int. It is not possible
+        # to use popen_ignore_int directly due to abscense of possibility
+        # to pass stdout=subprocess.PIPE argument in Popen call
```
@mbolivar-nordic 
Thanks for good advice! I've been reworked start_iceman according to it. Please check.

mbolivar-nordic replied:
Thanks for the update. The approach has a problem, however. You have preserved this commit as-is, then done the change I requested, then added a "fixup" commit. My original request was not to use a fixup commit, but instead:

> Please add a separate commit extending popen_ignore_int to take additional **kwargs which are passed on to the Popen constructor instead. Then in this commit, use the extended method, passing it any kwargs necessary.

That is:

1. rebase your second commit to be the first commit
2. squash the last commit into this commit, which will end up being the second commit

rikorsev commented:
Hi @mbolivar-nordic, @nashif, @kgugala, @pgielda,

Please, review the PR

Regards,
Alex

fkokosinski commented:
Hi @rikorsev! The changes from `boards/riscv` look good. I'm not leaving an approval though, because the majority of your changes affect twister, so you will need to have this PR approved by its maintainers.

mbolivar-nordic reviewed (changes requested):
Hi, zephyr policy is not to use "fixup" commits, so you need to rebase this pull request to adjust the commits as explained in more detail below.

pgielda commented:
@rikorsev those changes seem to be related more to west than riscv, so you have to mostly discuss this with @mbolivar-nordic

rikorsev commented:
Hi @mbolivar-nordic 

Please, review the PR and approve it if everything is fine.

Regards,
Alex

fkokosinski reviewed (approved):
LGTM for the RISC-V changes.
